ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/ConnectorUtils.java

```diff
@@ -313,4 +318,58 @@ public static SdkHttpFullRequest buildSdkRequest(
         }
         return builder.build();
     }
+
+    public static ConnectorAction createConnectorAction(Connector connector, ConnectorAction.ActionType actionType) {
+        Optional<ConnectorAction> batchPredictAction = connector.findAction(BATCH_PREDICT.name());
+        String predictEndpoint = batchPredictAction.get().getUrl();
+        Map<String, String> parameters = connector.getParameters() != null
+            ? new HashMap<>(connector.getParameters())
+            : Collections.emptyMap();
+
+        // Apply parameter substitution only if needed
+        if (!parameters.isEmpty()) {
+            StringSubstitutor substitutor = new StringSubstitutor(parameters, "${parameters.", "}");
+            predictEndpoint = substitutor.replace(predictEndpoint);
+        }
+
+        boolean isCancelAction = actionType == CANCEL_BATCH_PREDICT;
+
+        // Initialize the default method and requestBody
+        String method = "POST";
+        String requestBody = null;
+        String url = "";
+
+        switch (getRemoteServerFromURL(predictEndpoint)) {
+            case "sagemaker":
+                url = isCancelAction
+                    ? predictEndpoint.replace("CreateTransformJob", "StopTransformJob")
+                    : predictEndpoint.replace("CreateTransformJob", "DescribeTransformJob");
+                requestBody = "{ \"TransformJobName\" : \"${parameters.TransformJobName}\"}";
+                break;
+            case "openai":
+            case "cohere":
+                url = isCancelAction ? predictEndpoint + "/${parameters.id}/cancel" : predictEndpoint + "/${parameters.id}";
+                method = isCancelAction ? "POST" : "GET";
+                break;
+            case "bedrock":
+                url = isCancelAction
+                    ? predictEndpoint + "/${parameters.processedJobArn}/stop"
+                    : predictEndpoint + "/${parameters.processedJobArn}";
+                method = isCancelAction ? "POST" : "GET";
+                break;
```

Comment:
We should add the default branch for this switch statement to return null. In the GetTask and CancelTask, if the ConnectorAction is null, throw an exception with meaning logs like "please provide GetTask/CancelTask action in the connector".

Reply:
oh sorry I added it but I guess it got missed during refactoring. Let me add it